File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int max = 0, baseHeight = 0, area = 0, left = 0, right = height.length-1;
        while(left<right){
            area = Math.min(height[left], height[right]) * (right-left);
            max = Math.max(max, area);
            if(height[left]<=height[right]){
                left++;
            }
            else if(height[left]>height[right]){
                right--;
            }
        }
        return max;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxArea, l, r = 0, 0, len(height) - 1
        while l < r:
            area = (r - l) * min(height[l], height[r])
            maxArea = max(maxArea, area)
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return maxArea